runtests.py

```diff
@@ -1,317 +1,26 @@
-            elif a == '-a' or a == '--argument':
-                curlist = arglist
-            elif a == '-p' or a == '--pytest_arg':
-                curlist = pyt_arglist
-            # will also pass this option to pytest
-            elif a == '--lf':
-                lf = True
-            # will also pass this option to pytest
-            elif a == '--ff':
-                ff = True
-            elif a == '-l' or a == '--list':
-                list_only = True
-            elif a == '-c' or a == '--coverage':
-                coverage = True
-            elif a == '-h' or a == '--help':
-                usage(0)
-            else:
-                usage(1)
-        else:
-            curlist.append(a)
-            curlist = whitelist
-    if curlist is blacklist:
-        sys.exit('-x must be followed by a filter')
-    if curlist is arglist:
-        sys.exit('-a must be followed by an argument')
-    if curlist is pyt_arglist:
-        sys.exit('-p must be followed by an argument')
-    if lf and ff:
-        sys.exit('use either --lf or --ff, not both')
-    # empty string is a substring of all names
-    if not whitelist:
-        whitelist.append('')
-    if lf:
-        pyt_arglist.append('--lf')
-    if ff:
-        pyt_arglist.append('--ff')
-    if verbosity >= 1:
-        pyt_arglist.extend(['-v'] * verbosity)
-    elif verbosity < 0:
-        pyt_arglist.extend(['-q'] * (-verbosity))
-    if parallel_limit:
-        if '-n' not in pyt_arglist:
-            pyt_arglist.append('-n{}'.format(parallel_limit))
-
-    driver = Driver(whitelist=whitelist, blacklist=blacklist, lf=lf, ff=ff,
-                    arglist=arglist, pyt_arglist=pyt_arglist, verbosity=verbosity,
-                    parallel_limit=parallel_limit, xfail=[], coverage=coverage)
-
-    driver.prepend_path('PATH', [join(driver.cwd, 'scripts')])
-    driver.prepend_path('MYPYPATH', [driver.cwd])
-    driver.prepend_path('PYTHONPATH', [driver.cwd])
-
-    driver.add_flake8()
-    add_pytest(driver)
-
-    if list_only:
-        driver.list_tasks()
-        return
-
-    exit_code = driver.waiter.run()
-    t1 = time.perf_counter()
-    print('total runtime:', t1 - t0, 'sec')
+prog, *args = sys.argv
 
-    if verbosity >= 1:
-        times = driver.waiter.times2 if verbosity >= 2 else driver.waiter.times1
-        times_sortable = ((t, tp) for (tp, t) in times.items())
-        for total_time, test_type in sorted(times_sortable, reverse=True):
-            print('total time in %s: %f' % (test_type, total_time))
+cmds = {
+    'self': 'python3 -m mypy --config-file mypy_self_check.ini -p mypy',
+    'lint': 'flake8 -j0',
+    'pytest': 'pytest'
+}
 
-    sys.exit(exit_code)
+if not set(args).issubset(cmds):
+    print("usage:", prog, " ".join('[%s]' % k for k in cmds))
+    exit(1)
 
+if not args:
+    args = list(cmds)
 
-if __name__ == '__main__':
-    main()
+for arg in args:
+    cmd = cmds[arg]
+    print('$', cmd)
+    res = (system(cmd) & 0x7F00) >> 8
```

Comment:
Why the bit twiddling here? Wouldn't it be better to use `subrocess.call`?

Reply:
I want to have the full command copy-pastable in the source. To clarify that this is a shorthand script, and that nothing more is going on than a mere execution of a command.